### PR TITLE
Fix win-arm64 native varargs ABI

### LINKED ISSUE #89595: JIT: win-arm64 native varargs doesn't correctly split incoming vectors between register and stack
from the Runtime_71375.cs test source:
```
        // At the time of writing this test, the calling convention for incoming vector parameters on
        // Windows ARM64 was broken, so only the fact that "Problem" compiled without asserts was
        // checked. If/once the above is fixed, this test should be changed to actually call "Problem".
```

### PR DESCRIPTION
**Pull request by BruceForstall**
SIMD vector types should be passed in integer registers for win-arm64 native varargs. This might require splitting a Vector128 between x7 and stack.

See https://learn.microsoft.com/en-us/cpp/build/arm64-windows-abi-conventions?view=msvc-170#addendum-variadic-functions

Fixes #89595

### CODE REVIEW

**Comment by BruceForstall** [comment]
Only failure is known failure https://github.com/dotnet/runtime/issues/90593

**Comment by BruceForstall** [comment]
[No spmi asm diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=376964&view=ms.vss-build-web.run-extensions-tab) -- negligible TP change on arm64.

**Comment by BruceForstall** [comment]
@jakobbotsch PTAL
cc @dotnet/jit-contrib

**Comment by jkotas** [comment]
Does VM-side implementation of calling conventions (https://github.com/dotnet/runtime/blob/main/src/coreclr/vm/callingconvention.h) work as expected for this case?

File: src/coreclr/jit/compiler.hpp [hunk 1]
```diff
@@ -2688,6 +2688,12 @@ inline var_types Compiler::mangleVarArgsType(var_types type)
             default:
                 break;
         }
+
+        if (varTypeIsSIMD(type))
+        {
+            // Vectors also get passed in int registers. Use TYP_INT.
+            return TYP_INT;
```
Comment (jakobbotsch) [inline src/coreclr/jit/compiler.hpp:2695]: Is the size mismatch here fine?

**Review by jakobbotsch — APPROVED** [review]

File: src/coreclr/jit/compiler.hpp (hunk 1 above)
Comment (BruceForstall) [reply src/coreclr/jit/compiler.hpp:2695]: It turns out the size doesn't matter; the only thing that matters is INT (use general-purpose registers) or non-INT (use floaing-point/SIMD registers).

**Comment by BruceForstall** [comment]
> Does VM-side implementation of calling conventions (https://github.com/dotnet/runtime/blob/main/src/coreclr/vm/callingconvention.h) work as expected for this case?

It looks like it might already be covered:

https://github.com/dotnet/runtime/blob/64243bbf5e9ee53c0c4c5678f2cd8c7f1c9b4f6f/src/coreclr/vm/callingconvention.h#L1618-L1636

One thing that wasn't handled on the JIT is passing/receiving any Vector to/in win-arm64 varargs functions. How could I test this case through the callingconvention.h code?

**Comment by jkotas** [comment]
callingconvention.h is only used for GC stackroot reporting.

It may be useful to add a variant of the test with interleaving scalars and object refences in the vararg signature, so that we would catch the situation where the GC stackroot reporting does not report the varargs arguments correctly.

**Comment by BruceForstall** [comment]
> callingconvention.h is only used for GC stackroot reporting.
> 
> It may be useful to add a variant of the test with interleaving scalars and object refences in the vararg signature, so that we would catch the situation where the GC stackroot reporting does not report the varargs arguments correctly.

I stepped through callingconvention.h and saw the correct behavior occuring with the current code.

I added a few addtional test cases to the test.

File: src/tests/JIT/Regression/JitBlue/Runtime_71375/Runtime_71375.cs [hunk 2]
```diff
@@ -26,4 +24,58 @@ public static bool Problem()
...
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static int VarArgs4(int arg1, object arg2, int arg3, object arg4, int arg5, object arg6, Vector128<int> splitArg, __arglist) => splitArg.GetElement(0);
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static bool Case4()
+    {
+        // Spit the vector but also pass some object types so the GC needs to know about them.
+        return VarArgs4(0, new object(), 0, new object(), 0, new object(), Vector128<int>.AllBitsSet, __arglist()) != -1;
```
Comment (jkotas) [inline src/tests/JIT/Regression/JitBlue/Runtime_71375/Runtime_71375.cs:79]:
```suggestion
        return VarArgs4(0, new object(), 0, new object(), 0, new object(), Vector128<int>.AllBitsSet, __arglist(new object(), 1, new object())) != -1;
```
?

Comment (BruceForstall) [reply src/tests/JIT/Regression/JitBlue/Runtime_71375/Runtime_71375.cs]: looks good.

**Pull request merged**